bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/internal/discovery/ResolVBusBridgeDiscovery.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@Component(service = DiscoveryService.class)
+@NonNullByDefault
+public class ResolVBusBridgeDiscovery extends AbstractDiscoveryService {
+    private final Logger logger = LoggerFactory.getLogger(ResolVBusBridgeDiscovery.class);
+
+    public ResolVBusBridgeDiscovery() throws IllegalArgumentException {
+        super(ResolBindingConstants.SUPPORTED_BRIDGE_THING_TYPES_UIDS, 35, false);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.trace("Start discovery of Resol VBus-LAN Adapter");
+        scheduler.execute(searchRunnable);
+    }
+
+    /*
+     * The runnable for the search routine.
+     */
+    private Runnable searchRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            logger.trace("Start adapter discovery...");
+
+            try {
+                InetAddress broadcastAddress = InetAddress
+                        .getByAddress(new byte[] { (byte) 255, (byte) 255, (byte) 255, (byte) 255 });
+
+                TcpDataSource[] dataSources = TcpDataSourceProvider.discoverDataSources(broadcastAddress, 3, 500,
+                        false);
+
+                HashMap<String, TcpDataSource> currentDataSourceById = new HashMap<String, TcpDataSource>();
+                for (TcpDataSource ds : dataSources) {
+                    InetAddress address = ds.getAddress();
+                    String addressId = address.getHostAddress();
+                    TcpDataSource dsWithInfo;
+                    try {
+                        dsWithInfo = TcpDataSourceProvider.fetchInformation(address, 1500);
+                        logger.trace("Discovered Resol VBus-LAN interface @{} {} ({})", addressId,
+                                dsWithInfo.getDeviceName(), dsWithInfo.getSerial());
+
+                        currentDataSourceById.put(addressId, dsWithInfo);
+                        addAdapter(addressId, dsWithInfo);
+                        // here we can add the detection of Multi-Channel interfaces like DL3
+                    } catch (IOException ex) {
+                        /* address is no valid adapter */
+                    }
+
+                }
+            } catch (UnknownHostException e) {
+                logger.debug("Could not resolve IPv4 broadcast address");
+            }
+        }
+    };
+
+    private void addAdapter(String remoteIP, TcpDataSource dsWithInfo) {
+        String adapterSerial = dsWithInfo.getSerial();
+        Map<String, Object> properties = new HashMap<>(3);
+        properties.put("ipAddress", remoteIP);
+        properties.put("port", dsWithInfo.getLivePort());
+        properties.put("adapterSerial", adapterSerial);
+
+        ThingUID uid = new ThingUID(ResolBindingConstants.THING_TYPE_UID_BRIDGE, adapterSerial);
+        thingDiscovered(
+                DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(dsWithInfo.getName()).build());
```

Comment:
Can you add the representation property?

Reply:
The bridge is uniquely identified by the adapterSerial, which is read on discovery, but not required in file based configuration.
If I add a bridge manually in the configuration files the very same one is still autodetected, as long as I don't give the optional parameter adapterSerial. - Wouldn't it be better in this case to use the IP address as representation property?

Reply:
Not sure. I would go for the IP address.

Reply:
I though I read that it should be a not-changing property and the IP would even not be recommended. But 1. I don't find that statement anymore, 2 the scenario that a bridge is using DHCP and getting a changed IP is quite unlikely.